extensions/wikia/AdEngine/js/video/uapVideo.js

```diff
@@ -15,83 +18,115 @@ define('ext.wikia.adEngine.video.uapVideo', [
-			currentTime.style.width = '100%';
-		} else {
-			currentTime.style.width = '0';
-		}
+	function loadPorvata(params, slotContainer, providerContainer) {
+		params.container = slotContainer;
+
+		log(['VUAP loadPorvata', params], log.levels.debug, logGroup);
+
+		return porvata.inject(params)
+			.then(function (video) {
+				videoInterface.setup(video, [
+					'progressBar',
+					'pauseOverlay',
+					'volumeControl',
+					'closeButton',
+					'toggleAnimation'
+				], {
+					image: providerContainer,
+					container: slotContainer,
+					aspectRatio: params.aspectRatio,
+					videoAspectRatio: params.videoAspectRatio
+				});
+
+				video.addEventListener('allAdsCompleted', function () {
+					video.reload();
+				});
+
+				return video;
+			});
 	}
 
-	function addProgressListeners(video) {
-		video.addEventListener('start', updateProgressBar);
-		video.addEventListener('resume', updateProgressBar);
-		video.addEventListener('complete', updateProgressBar);
-		video.addEventListener('pause', function (ima) {
-			var progressBar = ima.container.querySelector('.progress-bar'),
-				currentTime = progressBar.querySelector('.current-time');
+	function loadPlaywire(params, adSlot, imageContainer) {
+		var container = doc.createElement('div');
+
+		container.classList.add('video-player', 'hidden');
+		adSlot.appendChild(container);
+
+		params.container = container;
+
+		log(['VUAP loadPlaywire', params], log.levels.debug, logGroup);
+		return playwire.inject(params)
+			.then(function (video) {
+				videoInterface.setup(video, [
+					'closeButton',
+					'toggleAnimation'
+				], {
+					image: imageContainer,
+					container: adSlot,
+					aspectRatio: params.aspectRatio,
+					videoAspectRatio: params.videoAspectRatio
+				});
+
+				video.addEventListener('wikiaAdStarted', function () {
+					var slotWidth = getSlotWidth(adSlot);
+					video.resize(slotWidth, getVideoHeight(slotWidth, params));
+				});
+				if (params.autoplay) {
+					var slotWidth = getSlotWidth(adSlot);
+					video.play(slotWidth, getVideoHeight(slotWidth, params));
+				}
 
-			currentTime.style.width = (currentTime.offsetWidth / progressBar.offsetWidth * 100) + '%';
-		});
+				return video;
+			});
 	}
 
-	function loadVideoAd(params, adSlot, imageContainer, getSlotWidth) {
-		getSlotWidth = getSlotWidth || defaultGetSlotWidth;
-
-		try {
-			var video = videoAdFactory.create(
-				getSlotWidth(adSlot),
-				getVideoHeight(getSlotWidth(adSlot), params),
-				adSlot,
-				{
-					src: params.src,
-					pos: params.slotName,
-					uap: uapContext.getUapId(),
-					passback: 'vuap'
-				}
-			);
-			addProgressListeners(video);
+	function loadVideoAd(params) {
+		var loadedPlayer,
+			providerContainer = adSlot.getProviderContainer(params.slotName),
+			slotContainer = providerContainer.parentNode,
+			videoWidth = getSlotWidth(slotContainer);
+
+		params.width = videoWidth;
+		params.height = getVideoHeight(videoWidth, params);
```

Comment:
we already pass to `getVideoHeight` width inside `params`. No ned to pass it as well as separate param